app/controllers/get_involved_controller.rb

```diff
@@ -88,4 +90,46 @@ def retrieve_consultation_outcomes
 
     Services.search_api.search(query)["results"]
   end
+
+  def time_until_closure(consultation)
+    days_left = (consultation["end_date"].to_date - Time.zone.now.to_date).to_i
+    case days_left
+    when :negative?.to_proc
+      t("get_involved.closed")
+    when :zero?.to_proc
+      t("get_involved.closing_today")
+    when 1
+      t("get_involved.closing_tomorrow")
+    else
+      t("get_involved.days_left", number_of_days: days_left)
+    end
+  end
+
+  def date_microformat(attribute_name)
+    attribute_name.to_date.strftime("%d %B %Y")
+  end
+
+  def filtered_links(array, close_status)
+    array.map do |item|
+      org_acronym = item["organisations"].map { |org|
+        org["acronym"]
+      }.join(", ")
+
+      org_time = item["organisations"].map { |org|
+        org["public_timestamp"]
+      }.join(", ")
+
+      {
+        link: {
+          text: item["title"],
+          path: item["link"],
+          description: "#{close_status} #{date_microformat(item['end_date'])}",
+        },
+        metadata: {
+          public_updated_at: Time.zone.parse(org_time),
+          document_type: org_acronym,
+        },
+      }
+    end
+  end
```

Comment:
Again just a suggestion but you could refactor the filtered_links to move the `.map` operation stuff in to it's own private method e.g.
```
  def filtered_links(array, close_status)
    array.map do |item|
      {
        link: {
          text: item["title"],
          path: item["link"],
          description: "#{close_status} #{date_microformat(item['end_date'])}",
        },
        metadata: {
          public_updated_at: Time.zone.parse(map_property(org["public_timestamp"])),
          document_type: map_property(org["acronym"]),
        },
      }
    end
  end

private

  def map_property(property)
    item["organisations"].map { |org|
      org[property]
    }.join(", ")
  end
end
```

Reply:
Like it - updated